### PR TITLE
Properties

### PR DESCRIPTION
**Pull request by Mariana-VV**

### CODE REVIEW

File: src/main/java/com/example/bookstore/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (sokimaaa) [inline src/main/java/com/example/bookstore/model/Book.java]: using @Data with entity is not recommended.

File: src/main/java/com/example/bookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Comment (sokimaaa) [inline src/main/java/com/example/bookstore/model/Book.java]: add constraints for fields asked by task, e.g. @Column(nullable = false)

File: src/main/java/com/example/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    public Book() {
+    }
```
Comment (sokimaaa) [inline src/main/java/com/example/bookstore/model/Book.java]: use @NoArgsConstructor from lombok to generate this constructor automatically

File: src/main/java/com/example/bookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                BigDecimal price, String description,
+                String coverImage) {
+        this.id = id;
+        this.title = title;
+        this.author = author;
+        this.isbn = isbn;
+        this.price = price;
+        this.description = description;
+        this.coverImage = coverImage;
+    }
```
Comment (sokimaaa) [inline src/main/java/com/example/bookstore/model/Book.java]: use @AllArgsConstructor from lombok to generate this constructor automatically and reduce code.

File: src/main/java/com/example/bookstore/repository/BookRepository.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import com.example.bookstore.model.Book;
+import java.util.List;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface BookRepository {
+    Book save(Book book);
+
+    List<Book> findAll();
+}
```
Comment (sokimaaa) [inline src/main/java/com/example/bookstore/repository/BookRepository.java]: @Repository is redundant here. 

```suggestion
public interface BookRepository {
    Book save(Book book);

    List<Book> findAll();
}
```

File: src/main/java/com/example/bookstore/model/Book.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (Rommelua) [inline src/main/java/com/example/bookstore/model/Book.java:13]:
```suggestion
```

**Review by Rommelua — APPROVED** [review]

**Pull request merged**